### PR TITLE
MAINT: pyproject.toml: Update required build dependencies

### PR DESCRIPTION
**Pull request by EwoutH**
Update setuptools, wheel and Cython to their latest versions.

### CODE REVIEW

**Comment by charris** [comment]
Why update setuptools? It is known to be a problem.

**Comment by EwoutH** [comment]
Just discovered that!

I was doing some experiments toward Python 3.12 support, but let's first wait for the second beta.

**Comment by charris** [comment]
> I was doing some experiments toward Python 3.12 support

We plan to switch to meson builds for 2.0, no setuptools, see #23838. We will then plan to backport the build changes for a 1.26 release, basically 1.25 with Python 3.12 support.

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -347,7 +347,7 @@ jobs:
         docker run --name the_container --interactive -v /:/host arm32v7/ubuntu:22.04 /bin/bash -c "
           apt update &&
           apt install -y git python3 python3-dev python3-pip  &&
-          python3 -m pip install cython==0.29.34 setuptools\<49.2.0 hypothesis==6.23.3 pytest==6.2.5 'typing_extensions>=4.2.0' &&
+          python3 -m pip install -U cython==0.29.35 setuptools hypothesis==6.23.3 pytest==6.2.5 'typing_extensions>=4.2.0' &&
```
Comment (mattip) [inline .github/workflows/build_test.yml:350]: Please leave the setuptools pin.

**Pull request closed without merging**